src/inmanta/protocol/rest/__init__.py

```diff
@@ -220,21 +220,35 @@ async def process(self) -> None:
             arg_type: Optional[type[object]] = self._argspec.annotations.get(arg)
             if arg in self._message:
                 # Argument is parameter in body of path of HTTP request
-                if (
-                    arg_type
-                    and self._properties.operation == "GET"
-                    and typing_inspect.is_generic_type(arg_type)
-                    and issubclass(typing_inspect.get_origin(arg_type), list)
-                    and not isinstance(self._message[arg], list)
-                    and len(typing_inspect.get_args(arg_type, evaluate=True)) == 1
-                    and isinstance(self._message[arg], typing_inspect.get_args(arg_type)[0])
-                ):
-                    # If a GET endpoint has a parameter of type list that is encoded as a URL query parameter and the
-                    # specific request provides a list with one element, urllib doesn't parse it as a list.
-                    # Map it here explicitly to a list.
-                    value = [self._message[arg]]
+                if self._properties.operation == "GET":
+                    # Check if the argument type is an Optional type (e.g., Optional[List[SomeType]])
+                    if typing_inspect.is_optional_type(arg_type):
+                        # Extract non-None types from the Optional type
+                        non_none_arg_types = [arg for arg in typing_inspect.get_args(arg_type) if arg is not type(None)]
+
+                        # If there's only one type and it's a list, simplify the type to just the list type
+                        if len(non_none_arg_types) == 1 and typing_inspect.get_origin(non_none_arg_types[0]) is list:
```

Comment:
Why not `issubclass(typing_inspect.get_origin(arg_type), list)` ?

Reply:
This:
`if len(non_none_arg_types) == 1 and issubclass(typing_inspect.get_origin(arg_type), list):`
doesn't work if that is what you mean?

Reply:
I don't know if this comment is still relevant as I reworked it quit a bit :p